### ISSUE TITLE
Add a priority signal apart from 'as'

### ISSUE THREAD

**Issue by yoavweiss** [w3c/preload#22]
Reading through a [comment](http://www.stevesouders.com/blog/2015/05/12/hero-image-custom-metrics/#comment-61106) on @stevesouders' blog, I realized that:
- The technique mentioned is likely to suffer from prioritization in current Chrome implementation. Since the image is not displayed, it won't get downloaded until other images are downloaded.
- An equivalent `preload` implementation would suffer from the same issues. A `preload` based image would download in a lower priority than discovered `<img>`s, as soon as layout hits.

Originally we talked about the author marking such images with an `as=style` or somesuch to tell the browser that this resource is high priority. But we [later](https://github.com/whatwg/fetch/issues/43) concluded that this is not the best way to go. We want `as` in order to set the right request headers, (and probably default priority), but we also want to be able to upgrade the priority of a certain resource request.

Should we add a `priority` attribute?

**Comment by igrigorik** [comment]
The scenario you're describing here is not specific to `preload`. In the linked thread they're talking about an image, but we could, just as well, be talking about any other asset type, the fetch for which may be initiated via HTML, CSS, JavaScript - e.g. for images alone we have `img`, `picture`, `url()`, `fetch()`, `image.src`, etc... And the key observation is that sometimes the user may want to explicitly override the priority (weight, dependency, or both) of a particular fetch - see [full rant](https://www.igvita.com/2014/10/02/extensible-web-resource-loading-manifesto/).

Some related threads:
- https://lists.w3.org/Archives/Public/public-whatwg-archive/2014Aug/0081.html - early proposal for exposing transport prioritization control to developers. At the time, HTTP/2 was still in flight and we didn't get far; we need to revisit that discussion.
- https://lists.w3.org/Archives/Public/public-webapps-github/2015Apr/0364.html - more recent thread.
- https://lists.w3.org/Archives/Public/www-style/2014Dec/0012.html
- https://www.w3.org/Bugs/Public/show_bug.cgi?id=26533#c1

Long story short, I think we need to resolve above discussions first. Once we do, the behavior you're looking for, and interop with `preload`, will fall out of them.

**Comment by yoavweiss** [comment]
So you propose to wait for the concept of `fetch-settings` (or the like) to land before we can upgrade a resource's priority?

**Comment by igrigorik** [comment]
@yoavweiss yes. A single "priority" attribute is not sufficient to address all of the use cases, and I'd rather we take the slightly longer route and solve the full problem via one consistent API - e.g. dependencies & weights, whether the fetch blocks load event, and so on.

**Comment by igrigorik** [comment]
Closing, feel free to reopen if you think fetch-settings (or whatever we end up calling it) is insufficient to address your use case.
